app/main.py

```diff
@@ -1,3 +1,67 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: int) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        elif isinstance(other, (int, float)):
+            return Distance(
+                km=self.km + other
+            )
+
+    def __iadd__(self, other: int) -> "Distance":
```

Comment:
The type hint for the `other` parameter should include `Distance` as well, i.e., `Union[Distance, int, float]`.